just_curious_game.ex

```diff
@@ -0,0 +1,325 @@
+  @spec get_state(%JustCurious.Game{}) :: atom
+  def get_state(game) do
+    hget("team:#{game.team}:game:#{game}", "state") |> String.to_atom
+  end
+
+  @spec set_state(%JustCurious.Game{}, atom) :: atom
+  def set_state(game, state) do
+    hset("team:#{game.team}:game:#{game}", "state", state)
+    :"#{state}"
+  end
+
+  @spec get_round(%JustCurious.Game{}) :: atom
+  def get_round(game), do: hget("team:#{game.team}:game:#{game}", "round") |> String.to_atom
+
+  @spec set_round(%JustCurious.Game{}, atom) :: atom
+  def set_round(game, new_round) do
+    hset("team:#{game.team}:game:#{game}", "round", new_round)
+    :"#{new_round}"
+  end
+
+  @spec get_chainlink(%JustCurious.Game{}) :: atom
+  def get_chainlink(game), do: hget("team:#{game.team}:game:#{game}", "chainlink") |> String.to_integer
+
+  @spec incr_chainlink(%JustCurious.Game{}) :: integer
+  def incr_chainlink(game), do: hincrby("team:#{game.team}:game:#{game}", "chainlink", 1)
+
+  ## TRANSITION HELPERS ##
+  @spec new?(%JustCurious.Game{}) :: boolean
+  def new?(game), do: get_state(game) == :new
+
+  @spec started?(%JustCurious.Game{}) :: boolean
+  def started?(game), do: not Enum.member?(~w(new dead)a, get_state(game))
+
+  @spec submitting?(%JustCurious.Game{}) :: boolean
+  def submitting?(game), do: get_round(game) == :submitting
+
+  @spec nominating?(%JustCurious.Game{}) :: boolean
+  def nominating?(game), do: get_round(game) == :nominating
+
+  @spec answering?(%JustCurious.Game{}) :: boolean
+  def answering?(game), do: get_round(game) == :answering
+
+  @spec complete?(%JustCurious.Game{}) :: boolean
+  def complete?(game), do: get_round(game) == :complete
+
+  @spec ended?(%JustCurious.Game{}) :: boolean
+  def ended?(game), do: get_state(game) == :dead
+
+  @spec failed?(%JustCurious.Game{}) :: boolean
+  def failed?(game), do: get_round(game) == :failed
+
+  @doc """
+  A unique string representing the game name and the team it belongs to
+  """
+  @spec uuid(%JustCurious.Game{}) :: String.t
+  def uuid(game), do: "#{game.team}_#{game}"
+
+  @doc """
+  Adds a user to the game under the specified `kind` and removes them from any
+  other kind they might be on. If the user is not already a participant, they
+  are assigned a pseudonym for the game.
+  """
+  @spec add_participant(%JustCurious.Game{}, String.t, atom) :: :ok
+  def add_participant(game, user, kind \\ :player) do
+    if user != jc_bot_id(game.team) do # Just Curious can't participate
+      game_path = "team:#{game.team}:game:#{game}"
+
+      ~w(player spectator invite join)a |> Enum.each(&(srem("#{game_path}:#{&1}s", user)))
+
+      existing_pseudonym = get_pseudonym(game, user)
+      pseudonym          = if existing_pseudonym == :undefined, do: generate_name(game.team, :pseudonym), else: existing_pseudonym
+
+      sadd("#{game_path}:#{kind}s", user)
+      hset("#{game_path}:participant:#{user}", "pseudonym", pseudonym)
+    end
+
+    :ok
+  end
+
+  ## PARTICIPANTS ##
```

Comment:
I like these bits of organization via comments! They seem to be just enough to add some direction to the code without getting in the way.